From 4e45b69028d95d2f8fe693733d9250eb57753533 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 28 Jan 2021 08:16:47 +0100
Subject: [PATCH 1/1] test: correct entry point to pytest
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With Pytest 6.0.2 'make tests' fails:

sandbox: Traceback (most recent call last):
  File "./test/py/test.py", line 20, in <module>
    sys.exit(load_entry_point('pytest', 'console_scripts', 'pytest')(args))
TypeError: console_main() takes 0 positional arguments but 1 was given

The definition of console_scripts has changed as follows:

Pytest 4.6.1:

[options.entry_points]
console_scripts =
        pytest=pytest:main
        py.test=pytest:main
        
Pytest 6.0.2:
        
[options.entry_points]
console_scripts =
    pytest=pytest:console_main
    py.test=pytest:console_main

The new function console_main() has a comment:
"This function is not meant for programmable use; use `main()`"

Hence let's call pytest.main() directly.
Move args processing into the __main__ paragraph.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
This patch is enough to run the Python tests both on Gitlab and on a local
Debian Bullseye with Pytest 6.0.2.

But the output on Pytest 6.0.2 is while running incomplete: The names of
the tests are not written:

.....s.....sss.ssssss................ssssssssssssssssssssssssssss........
.s.......................................................................

The detailed summary at the end of the test is complete.

So we will need follow up patches.
---
 test/py/test.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/test/py/test.py b/test/py/test.py
index bee88d96bc..285fda5425 100755
--- a/test/py/test.py
+++ b/test/py/test.py
@@ -10,11 +10,11 @@
 import os
 import os.path
 import sys
+import pytest
 from pkg_resources import load_entry_point
 
-# argv; py.test test_directory_name user-supplied-arguments
-args = [os.path.dirname(__file__) + '/tests']
-args.extend(sys.argv)
-
 if __name__ == '__main__':
-    sys.exit(load_entry_point('pytest', 'console_scripts', 'pytest')(args))
+    # argv; py.test test_directory_name user-supplied-arguments
+    args = [os.path.dirname(__file__) + '/tests']
+    args.extend(sys.argv)
+    sys.exit(pytest.main(args))
-- 
2.29.2

